pkg/virt-launcher/virtwrap/access-credentials/access_credentials.go

```diff
@@ -597,3 +498,86 @@ func (l *AccessCredentialManager) HandleQemuAgentAccessCredentials(vmi *v1.Virtu
 
 	return nil
 }
+
+type accessCredentialsInfo struct {
+	// secret name mapped to authorized_keys in that secret
+	secretMap map[string][]string
+	// filepath mapped to secretNames
+	userSSHMap map[string][]string
+	// maps users to passwords
+	userPasswordMap map[string]string
+}
+
+func (a *accessCredentialsInfo) addAccessCredential(accessCred *v1.AccessCredential) error {
+	secretName := getSecret(accessCred)
+	if secretName == "" {
+		return nil
```

Comment:
Should we here return an error? Or it is not consider a problematic situation to not have the secret?

Reply:
It's not an error. If user does not specify a secret, then there is no key to add to the accumulation struct.

Reply:
Got it! Thanks.